patches/minecraft/net/minecraft/client/gui/DisplayEffectsScreen.java.patch

```diff
@@ -8,19 +8,19 @@
           this.field_147003_i = 160 + (this.width - this.field_146999_f - 200) / 2;
           this.field_147045_u = true;
        }
-@@ -84,6 +85,7 @@
-       int i = this.field_147009_r;
+@@ -59,7 +60,7 @@
+             j = 132 / (collection.size() - 1);
+          }
  
-       for(EffectInstance effectinstance : p_214077_3_) {
-+         if (!effectinstance.func_188419_a().shouldRender(effectinstance)) continue;
-          Effect effect = effectinstance.func_188419_a();
-          blit(p_214077_1_ + 6, i + 7, this.blitOffset, 18, 18, potionspriteuploader.func_215288_a(effect));
-          i += p_214077_2_;
-@@ -95,6 +97,9 @@
+-         Iterable<EffectInstance> iterable = Ordering.<EffectInstance>natural().sortedCopy(collection);
++         Iterable<EffectInstance> iterable = collection.stream().filter( effectInstance -> effectInstance.func_188419_a().shouldRender(effectInstance)).sorted().collect(java.util.stream.Collectors.toList());
```

Comment:
This should
- keep the same order (EffectInstance is Comparable)
- result in the same kind of list (a new mutable new ArrayList)